src/scraper_xml.cpp

```diff
@@ -408,52 +408,34 @@ void FeListXMLParser::start_element(
 					(*m_itr).set_info( FeRomInfo::Control, attribute[i+1] );
 			}
 		}
-		else if ( strcmp( element, "display" ) == 0 )
-		{
-			for ( int i=0; attribute[i]; i+=2 )
-			{
-				if ( strcmp( attribute[i], "rotate" ) == 0 )
-					(*m_itr).set_info( FeRomInfo::Rotation, attribute[i+1] );
-				else if ( strcmp( attribute[i], "type" ) == 0 )
-					(*m_itr).set_info( FeRomInfo::DisplayType, attribute[i+1] );
-			}
-			m_displays++;
-		}
-		else if ( strcmp( element, "driver" ) == 0 )
-		{
-			for ( int i=0; attribute[i]; i+=2 )
-			{
-				if ( strcmp( attribute[i], "status" ) == 0 )
-				{
-					(*m_itr).set_info( FeRomInfo::Status, attribute[i+1] );
-					break;
-				}
-			}
-		}
 		else if ( strcmp( element, "control" ) == 0 )
-		{
+		{			
 			std::string type, ways, old_type;
-
+			
 			old_type = (*m_itr).get_info( FeRomInfo::Control );
 			if ( !old_type.empty() )
 				old_type += ",";
-
+			
 			for ( int i=0; attribute[i]; i+=2 )
 			{
 				if ( strcmp( attribute[i], "type" ) == 0 )
 					type = attribute[i+1];
 				else if ( strcmp( attribute[i], "ways" ) == 0 )
 					ways = attribute[i+1];
+				
+				if (( strcmp( attribute[i], "buttons" ) == 0 )
+						&& (*m_itr).get_info( FeRomInfo::Buttons ).empty() )
+					(*m_itr).set_info( FeRomInfo::Buttons, attribute[i+1] );
```

Comment:
Hi, thanks for this, I am just trying to review your pull request now.  Am I correct in understanding that the three lines inserted here are what have fixed this issue (lines 426 to 428)?

Are any of the other edits in the file relevant to your fix (they don't appear to be)?

Reply:
@mickelson True. Sorry about the issues with line endings and spaces. I just realized it made this commit a mess.